source/hic_hal/freescale/kl26z/uart.c

```diff
@@ -283,9 +237,18 @@ void UART_RX_TX_IRQHandler(void)
         if ((s1 & UART_S1_NF_MASK) || (s1 & UART_S1_FE_MASK)) {
             errorData = UART->D;
         } else {
-            read_buffer.data[read_buffer.idx_in++] = UART->D;
-            read_buffer.idx_in &= (UART_BUFFER_SIZE - 1);
-            read_buffer.cnt_in++;
+            uint32_t free;
+            uint8_t data;
+            
+            data = UART1->D;
+            free = circ_buf_count_free(&read_buffer);
+            if (free > RX_OVRF_MSG_SIZE) {
+                circ_buf_push(&read_buffer, data);
+            } else if (RX_OVRF_MSG_SIZE == free) {
+                circ_buf_write(&read_buffer, (uint8_t*)RX_OVRF_MSG, RX_OVRF_MSG_SIZE);
```

Comment:
There are a few targets which do support and use hardware flow control, such as the nrf51-dk. Most, like the K64F don't though.

DAPLink does throttle USB CDC data coming from the host PC, so DAPLink's buffers in this direction will never overflow. The problem occurs when data is going from DAPLink to the PC. If the PC stops reading data from DAPLink then DAPLink's buffers will fill. If DAPLink can't stop the target from sending more data (as is the case in targets without flow control) then eventually DAPLink's buffers will overflow.